esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+The `@name` attribute is used for tracing and error reporting. The `@target` attribute is an xpath specifying the default root node of the target for all subsequent operations. If it is not present the root is the `<soap:Envelope>`. The `{$query}` and `{$request}` variables are special system macros that are set to the name of the backend query and reqeust respectively.
+
+The `@xmlns` attribute is the standard XML definition of a namespace prefix. For ESDL scripts the namespace URN `urn:hpcc:esdl:script` is recommended but not required. You can also define your own prefix (we use `es`), but the namespace prefixes and definitions should be consistent across all scripts in a service.
+
+```xml
+<es:param name="ExtendedSearchEnable" value="0"/>
+```
+
+This line defines a parameter named `ExtendedSearchEnable` with a default value of zero `0`. To make use of one of the values that the scripting engine makes available, a script must declare that parameter with the `<es:param>` command. Setting the  `@value` attribute is important to ensure you don't trigger an error by referencing an undefined parameter. This parameter is pulled from the Authorization results. It's name corresponds to the _path_ value of a _Secure User Setting_. These are configured using configmgr under an EspProcess's _ESP Service Bindings_ tab.
+
+```xml
+<es:source xpath="soap:Body/*/*/Row/Options/StartDate">
+```
+
+This modifies the default root node of the source of all child operations. The `@xpath` attribute speficies the node to make the default source root node.
+
+```xml
+<es:set-value target="Options/StartDate" select="Year"/>
+```
+
+This sets the value of the target node specified by `@target` to the value of the node specified by `@source`. Any previous value of `@target` is replaced by the new value of `@source`. Because this command is a child of the previous `<es:source>` command, the `@select` is relative to `soap:Body/*/*/Row/Options/StartDate`
+
+```xml
+<es:append-to-value target="Options/StartDate" select="Month"/>
+<es:append-to-value target="Options/StartDate" select="Day"/>
+```
+
+These both append the value of the `@select` node to the value of the `@target` node.
+
+## Transaction Flow and Entry Points
+
+Which version reads better?
```

Comment:
The table doesn't read well as a substitute for a flowchart. It doesn't cleanly support the branching logic related to the service entry point, and PreLogging will be deprecated and introduce more conditional logic.

As for the flowchart, there are options for building UML diagrams (which can represent flowcharts) in markdown syntax. It might be worth exploring these options for future updates.

Reply:
Ok. I've removed the table and used mermaid markup for the flowchart. 

That's a great suggestion to use markup. I thought I'd tried to find something at the time, but either mermaid wasn't supported natively by GitHub when I committed this or I'd missed it.